src/main/java/mate/academy/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -25,6 +29,17 @@ public List<BookDto> getAllBooks() {
                 .toList();
     }
 
+    @Override
+    public List<BookDto> searchBooks(BookSearchParametersDto params) {
+        Specification<Book> byTitle =
```

Comment:
what if you have 10 search parameters? 
Let's create SpecificationBuilder and SpecificationProvider for each parameter
or you can you BookSpecificationProvider with methods that return particular specificaiton